### PR TITLE
Add `ruff` linter config to `pyproject.toml`

### PR DESCRIPTION
**Pull request by tkoyama010**
### Overview

We previously considered introducing ruff (https://github.com/astral-sh/ruff) and dismissed it (https://github.com/pyvista/pyvista/pull/4115). On the other hand, the `matplotlib` project has introduced ruff (https://github.com/matplotlib/matplotlib/pull/25147) and `numpy` project is also considering using it. ruff has a linter and a formatter, Last time I tried to introduce both of those, I failed. So this time I would like to propose only the linter which has less impact.

### Details

- None

### CODE REVIEW

File: pyvista/trame/ui/base_viewer.py [hunk 1]
```diff
@@ -8,7 +8,6 @@
 import io
 
 from trame.app import get_server
-from trame.widgets import html
```
Comment (tkoyama010) [inline pyvista/trame/ui/base_viewer.py:11]: I am not familiar with trame viewer so I don't know if this import makes sense.

Comment (banesullivan) [reply pyvista/trame/ui/base_viewer.py:11]: cc @annehaley or @alesgenova

Comment (annehaley) [reply pyvista/trame/ui/base_viewer.py:11]: This is fine to remove, this import is unused.

**Review by banesullivan — APPROVED** [review]
Lgtm, thanks for making this happen! Excited to be using ruff 🐶

**Pull request merged**